src/hotspot/share/cds/filemap.cpp

```diff
@@ -221,31 +223,52 @@ void FileMapInfo::populate_header(size_t core_region_alignment) {
     }
     FREE_C_HEAP_ARRAY(const char, default_base_archive_name);
   }
+  ResourceMark rm;
+  GrowableArray<const char*>* app_cp_array = create_dumptime_app_classpath_array();
+  int len = app_cp_array->length();
+  char* lcp = NULL;
+  if (len > 0) {
+    lcp = longest_common_app_classpath(len, app_cp_array);
+  }
+  if (lcp != NULL) {
+    longest_common_prefix_size = strlen(lcp) + 1;
+    header_size += longest_common_prefix_size;
+    longest_common_prefix_offset = c_header_size + base_archive_name_size;
+  }
   _header = (FileMapHeader*)os::malloc(header_size, mtInternal);
   memset((void*)_header, 0, header_size);
   _header->populate(this,
                     core_region_alignment,
                     header_size,
                     base_archive_name_size,
-                    base_archive_name_offset);
+                    base_archive_name_offset,
+                    longest_common_prefix_size,
+                    longest_common_prefix_offset,
+                    (const char*)lcp);
 }
 
 void FileMapHeader::populate(FileMapInfo *info, size_t core_region_alignment,
                              size_t header_size, size_t base_archive_name_size,
-                             size_t base_archive_name_offset) {
+                             size_t base_archive_name_offset, size_t common_app_classpath_size,
+                             size_t common_app_classpath_offset, const char* lcp) {
   // 1. We require _generic_header._magic to be at the beginning of the file
   // 2. FileMapHeader also assumes that _generic_header is at the beginning of the file
   assert(offset_of(FileMapHeader, _generic_header) == 0, "must be");
   set_header_size((unsigned int)header_size);
   set_base_archive_name_offset((unsigned int)base_archive_name_offset);
   set_base_archive_name_size((unsigned int)base_archive_name_size);
+  set_common_app_classpath_offset((unsigned int)common_app_classpath_offset);
+  set_common_app_classpath_size((unsigned int)common_app_classpath_size);
```

Comment:
I think there's no need to record the actual `common_app_classpath` string into the `FileMapHeader`. It should be enough to just store the length of the longest common path prefix. E.g., at runtime, the function `FileMapInfo::check_paths_ignoring_common_path` currently only uses `strlen(dumptime_prefix)`.

Also, `longest_common_app_classpath` can just return the length instead of the string.

Reply:
I've made the change. The changeset is smaller now.